trieChecker/cmd/main.go

```diff
@@ -0,0 +1,300 @@
+	"github.com/urfave/cli"
+)
+
+const (
+	defaultLogsPath      = "logs"
+	logFilePrefix        = "trie-checker"
+	maxTrieLevelInMemory = 5
+	rootHashLength       = 32
+	addressLength        = 32
+	maxDirs              = 100
+)
+
+var log = logger.GetOrCreate("main")
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "Relay CLI app"
+	app.Usage = "This is the entry point for the tool that checks the trie DB"
+	app.Flags = getFlags()
+	app.Authors = []cli.Author{
+		{
+			Name:  "The Elrond Team",
+			Email: "contact@elrond.com",
+		},
+	}
+
+	app.Action = func(c *cli.Context) error {
+		return startProcess(c)
+	}
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+		return
+	}
+
+	log.Info("finished processing trie")
+}
+
+func startProcess(c *cli.Context) error {
+	flagsConfig := getFlagsConfig(c)
+
+	_, errLogger := attachFileLogger(log, flagsConfig)
+	if errLogger != nil {
+		return errLogger
+	}
+
+	log.Info("sanity checks...")
+
+	err := logger.SetLogLevel(flagsConfig.LogLevel)
+	if err != nil {
+		return err
+	}
+
+	rootHash, err := hex.DecodeString(flagsConfig.HexRootHash)
+	if err != nil {
+		return fmt.Errorf("%w when decoding the provided hex root hash", err)
+	}
+	if len(rootHash) != rootHashLength {
+		return fmt.Errorf("wrong root hash length: expected %d, got %d", rootHashLength, len(rootHash))
+	}
+
+	maxDBValue, err := getMaxDBValue(filepath.Join(flagsConfig.WorkingDir, flagsConfig.DbDir))
+	if err != nil {
+		return err
+	}
+
+	log.Info("starting processing trie", "pid", os.Getpid())
+
+	return checkTrie(flagsConfig, rootHash, maxDBValue)
+}
+
+func attachFileLogger(log logger.Logger, flagsConfig config.ContextFlagsConfig) (elrondFactory.FileLoggingHandler, error) {
+	var fileLogging elrondFactory.FileLoggingHandler
+	var err error
+	if flagsConfig.SaveLogFile {
+		fileLogging, err = logging.NewFileLogging(flagsConfig.WorkingDir, defaultLogsPath, logFilePrefix)
+		if err != nil {
+			return nil, fmt.Errorf("%w creating a log file", err)
+		}
+	}
+
+	err = logger.SetDisplayByteSlice(logger.ToHex)
+	log.LogIfError(err)
+	logger.ToggleLoggerName(flagsConfig.EnableLogName)
+	logLevelFlagValue := flagsConfig.LogLevel
+	err = logger.SetLogLevel(logLevelFlagValue)
+	if err != nil {
+		return nil, err
+	}
+
+	if flagsConfig.DisableAnsiColor {
+		err = logger.RemoveLogObserver(os.Stdout)
+		if err != nil {
+			return nil, err
+		}
+
+		err = logger.AddLogObserver(os.Stdout, &logger.PlainFormatter{})
+		if err != nil {
+			return nil, err
+		}
+	}
+	log.Trace("logger updated", "level", logLevelFlagValue, "disable ANSI color", flagsConfig.DisableAnsiColor)
+
+	return fileLogging, nil
+}
+
+func getMaxDBValue(parentDir string) (int, error) {
+	contents, err := ioutil.ReadDir(parentDir)
+	if err != nil {
+		return 0, err
+	}
+
+	directories := make([]string, 0)
+	for _, c := range contents {
+		if c.IsDir() {
+			_, ok := big.NewInt(0).SetString(c.Name(), 10)
+			if !ok {
+				continue
```

Comment:
log warns at least?

Reply:
added log.Debug. This can happen when we work with temp directories for easy switch between data sets